virttest/utils_zcrypt.py

```diff
@@ -0,0 +1,378 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Sebastian Mitterle <smitterl@redhat.com>
+
+"""
+Module for crypto device management on s390x aka IBM Z
+
+Example:
+
+    # get available domains and set up host
+    lszcrypt_info = CryptoDeviceInfoBuilder.get()
+    mask_helper = APMaskHelper.from_infos(lszcrypt_info.domains)
+    matrix_device = MatrixDevice.from_infos(lszcrypt_info.domains)
+    ...
+    # Use matrix_device.uuid for guest definition/device attachment
+    # <hostdev ...><source><address uuid="$uuid"></source></hostdev>
+    ...
+    # restore host gracefully
+    matrix_device.unassign_all()
+    mask_helper.return_to_host_all()
+
+"""
+from uuid import uuid1
+from os.path import join
+
+from avocado.utils import process
+from virttest.utils_misc import cmd_status_output
+
+# timeout value in seconds for any command run
+CMD_TIMEOUT = 60
+# for debugging, set True to get verbose command run output
+VERBOSE = False
+# lszcrypt output if there are not crypto devices
+NO_DEVICES = "No crypto card devices found"
+
+
+class CryptoDeviceInfoBuilder(object):
+    """
+    Represents the available crypto device info
+    """
+
+    def __init__(self, session=None):
+        """
+        Initializes instance
+
+        :param session: guest session; if None, host info is handled
+        """
+        self.session = session
+
+    def get_info(self):
+        """
+        Gets crypto device info
+
+        :return: CryptoDeviceInfo instance
+        """
+        info = CryptoDeviceInfo()
+        e, o = cmd_status_output("lszcrypt -V", shell=True,
```

Comment:
e,o can be more meaningful variables rather than single letter

Reply:
@chunfuwen Yep, I changed to "err, out".